tests/test_unit/approval_client/client.py

```diff
@@ -120,15 +118,22 @@ def id(
             output.append(id_)
         return output
 
+    @staticmethod
+    def create_external_id(id: int | Sequence[int]) -> str | list[str]:
+        """Creates an external ID from an ID."""
+        if isinstance(id, int):
+            return f"external_{id}"
+        return [f"external_{i}" for i in id]
+
     def external_id(
         self,
         id: int | Sequence[int],
     ) -> str | list[str]:
         try:
             return self._reverse_cache[id] if isinstance(id, int) else [self._reverse_cache[i] for i in id]
         except KeyError:
-            if self._generate_random_external_id_on_lookup_failure:
-                return "".join(random.choices(string.ascii_letters + string.digits, k=10))
+            if self._allow_reverse_lookup:
+                return self.create_external_id(id) if isinstance(id, int) else [self.create_external_id(i) for i in id]
```

Comment:
This ensure that when we call this multiple times we get the same external ID. This is needed in the test of dumping extraction pipelines.